PDU/Helper.js

```diff
@@ -129,12 +129,35 @@ Helper.decode7Bit = function(text, inLen, alignBits)
         if(digit % 128 == 27){
             inExt = true;
         } else {
-            if(inExt){
-                ret.push(Helper.EXTENDED_TABLE.charCodeAt(digit));
-                inExt = false;
-            } else {
-                ret.push(Helper.ALPHABET_7BIT.charCodeAt(digit));
-            }
+          var c;
+          if(inExt){
+              c = Helper.EXTENDED_TABLE.charCodeAt(digit);
+              inExt = false;
+          } else {
+              c = Helper.ALPHABET_7BIT.charCodeAt(digit);
+          }
+          if (c < 0x80) {
+              ret.push(c);
+          } else if (c < 0x800) {
+              ret.push(0xC0 | (c >> 6), 0x80 | (c & 0x3F));
+          } else if (
+              ((c & 0xFC00) == 0xD800) && (digit + 1) < Helper.EXTENDED_TABLE.length &&
+              ((Helper.EXTENDED_TABLE.charCodeAt(digit + 1) & 0xFC00) == 0xDC00)) {
+              // Surrogate Pair
+              c = 0x10000 + ((c & 0x03FF) << 10) + (str.charCodeAt(++digit) & 0x03FF);
```

Comment:
Where does the variable `str` come from? 🤔

Reply:
uups... in fact you are right ... interestingly this never had an issue in my tests :-)

Reply:
@lipflip how it should be correctly?

Reply:
I assume str needs to be "Helper.EXTENDED_TABLE", makes that sense?

Reply:
I guess

Reply:
Then two guesses is best :-)  I also have no idea in which cases this fallback would get active ... ideally we have a test for that, but thats not that simple :-)